tests/src/test/java/com/maxio/advancedbilling/controllers/subscriptionnotes/SubscriptionNotesControllerListTest.java

```diff
@@ -0,0 +1,133 @@
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.maxio.advancedbilling.utils.RequestUtils.createNoteBody;
+import static com.maxio.advancedbilling.utils.assertions.CommonAssertions.assertNotFound;
+import static com.maxio.advancedbilling.utils.assertions.CommonAssertions.assertThatErrorListResponse;
+import static com.maxio.advancedbilling.utils.assertions.CommonAssertions.assertUnauthorized;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SubscriptionNotesControllerListTest {
+    private final TestSetup testSetup = new TestSetup();
+    private final AdvancedBillingClient client = TestClientProvider.getClient();
+    private final SubscriptionNotesController subscriptionNotesController = client.getSubscriptionNotesController();
+    private final List<SubscriptionNoteResponse> notes = new ArrayList<>();
+    private Product product;
+    private Customer customer;
+    private Subscription subscription;
+
+    @BeforeAll
+    void setUp() throws IOException, ApiException {
+        ProductFamily productFamily = testSetup.createProductFamily();
+        product = testSetup.createProduct(productFamily);
+        customer = testSetup.createCustomer();
+        subscription = testSetup.createSubscription(customer, product);
+        for (int i = 0; i < 10; ++i) {
+            notes.add(subscriptionNotesController
+                    .createSubscriptionNote(subscription.getId(), createNoteBody("Test note " + i, i % 2 == 0)));
+        }
+    }
+
+    @AfterAll
+    void teardown() throws IOException, ApiException {
+        new TestTeardown().deleteCustomer(customer);
+    }
+
+    @Test
+    void shouldListAllNotes() throws IOException, ApiException {
+        // when
+        List<SubscriptionNoteResponse> response = subscriptionNotesController.listSubscriptionNotes(new ListSubscriptionNotesInput.Builder()
+                .subscriptionId(subscription.getId()).build()
+        );
+
+        // then
+        assertThat(response)
+                .hasSize(10)
+                .usingRecursiveFieldByFieldElementComparator()
+                .containsExactlyInAnyOrderElementsOf(notes);
+    }
+
+    @Test
+    void shouldListNotesWithPaging() throws IOException, ApiException {
+        // when
+        List<SubscriptionNoteResponse> page1 = subscriptionNotesController
+                .listSubscriptionNotes(new ListSubscriptionNotesInput(subscription.getId(), 1, 2));
+        List<SubscriptionNoteResponse> page2 = subscriptionNotesController
+                .listSubscriptionNotes(new ListSubscriptionNotesInput(subscription.getId(), 2, 2));
+        List<SubscriptionNoteResponse> bigPage = subscriptionNotesController
```

Comment:
i meant smth like 
```
List<SubscriptionNoteResponse> bigPage2 = subscriptionNotesController.listSubscriptionNotes(new ListSubscriptionNotesInput(subscription.getId(), 2, 8));
 assertThat(bigPage2)
                .hasSize(2);
Stream.concat(bigpage.stream(), bigPage2.stream()).collect(Collectors.toSet()) .. .usingRecursiveFieldByFieldElementComparator()
                .containsExactlyInAnyOrderElementsOf(notes);
```
but current one works too

I was just wondering about the comment `order is not deterministic, so cannot assert specific notes`, so I was thinking, is order different every time we do listing? then paging would break

Reply:
Probably order is consistent across pages. The current solution tests already that second page contains different notes so it's ok.